Software

Gautier30 opened the pull request.

ReikoR committed:
Initialize software

ReikoR committed:
Initialize software

ReikoR committed:
Merge branch 'software' of github.com:ut-robotics/picr22-team-robodendron into software

Gautier30 committed:
Add files via upload

Akustav commented on software/follow_ball_depth.py:
```diff
@@ -0,0 +1,381 @@
+    """
+    return speed
+
+def speed_limitation(speed,direction):
```
If you need hard limiters write it in the driving code that us responsive for omni motion

Akustav commented on software/follow_ball_depth.py:
```diff
@@ -0,0 +1,381 @@
+from realsense_depth import *
+
+exitFlag = 0
+global speedz
```
Globals should not be used. Try to utilize class variables

Akustav commented on software/follow_ball_depth.py:
```diff
@@ -0,0 +1,381 @@
+                print(msg.get("signal"))
+
+            else:
+                for robot in range(len(msg.get("targets"))):
```
Try to use a if `msg.get("targets").index(robot_id) `here to find the index of your robots ID.

Akustav commented on software/follow_ball_depth.py:
```diff
@@ -0,0 +1,381 @@
+
+
+    def align_on(self):
+        global state
```
Globals again

Akustav commented on software/follow_ball_depth.py:
```diff
@@ -0,0 +1,381 @@
+    
+
+def main_loop():
+    global speedz
```
Instead of globals utilize a Class or similar that can be passed where needed

Akustav commented on software/follow_ball_depth.py:
```diff
@@ -0,0 +1,381 @@
+
+    try:
+        while True:
+            state = 'find_ball'
```
Utilize a enum instead of strings. Is more robust

Akustav reviewed (commented):
A lot of globals and try to utilize enums or similar instead of simple trings for state identification. Try to fix that

Gautier30 committed:
Add files via upload

We corrected the issues that were raised on the previous pull request (no more global variables, enums instead of strings for states, correction of the CamelCase/snake_case formatting...). Awaiting further feedback.

Gautier30 committed:
Delete README.asciidoc

This README file is not in use, refer to README.md instead.

Gautier30 committed:
Add files via upload

Akustav commented on software/follow_ball_depth.py:
```diff
@@ -0,0 +1,300 @@
+                
+                if largest:
+        
+                    robot.speedz = (processedData.debug_frame.shape[1]/2) - largest.x 
```
Kinda in-intuitive to get frame size from the debug frame. If you need to use the sizes, I suggest putting this data into the processedData

Akustav reviewed (commented):
Ref module reconnecting is questionable and I suggest streamlining where you get the frames resolution.

Gautier30 committed:
Updated the software documentation

Gautier30 committed:
Updated the software documentation

Gautier30 committed:
Updated the software documentation

ReikoR commented on software/requirements.txt:
```diff
@@ -0,0 +1,8 @@
+pandas==1.5.3
+pyrealsense2==2.53.1.4623
+pyserial==3.5
+segment==0.2.0
```
`segment` is from [boot camp code](https://github.com/ut-robotics/picr22-boot-camp-programming), not this package: https://pypi.org/project/segment/.
It can't be installed with pip.

Gautier30 replied:
My bad, I didn't see it after running the pipreqs command to generate the requirements.txt... It's fixed now

ReikoR replied:
`segment` module from boot camp code still needs installation. Instructions should be added to README.

Gautier30 replied:
I've added a few lines to the documentation about this module. Please let me know if it works with you.

ReikoR replied:
After cloning you can follow the installation instructions in the [boot camp code README](https://github.com/ut-robotics/picr22-boot-camp-programming/blob/main/README.md):
> How to use
>Segmentation module installation:
>```
>cd segment_module
>pip3.9 install .
>```

There is no need to move the files around.

Gautier30 replied:
Is it better now?

Gautier30 committed:
Updated the software documentation

Gautier30 committed:
Updated the software documentation

Gautier30 committed:
Updated the software documentation

Gautier30 committed:
Updated the software documentation

ReikoR reviewed (changes requested):
Documentation is ok, but there are unresolved code issues.
